app/scripts/lib/snap-keyring/snap-keyring.ts

```
@@ -145,117 +233,137 @@ export const snapKeyringBuilder = (
+              // User has clicked on "OK"
+              trackSnapAccountEvent(
+                MetaMetricsEventName.AddSnapAccountSuccessClicked,
               );
             }
-          } else {
-            // User has cancelled account creation
-            await handleUserInput(confirmationResult);
 
-            throw new Error('User denied account creation');
-          }
-        } finally {
-          // We do not have a `else` clause here, as it's used if the request was
-          // canceled by the user, thus it's not a "fail" (not an error).
-          if (confirmationResult) {
             trackSnapAccountEvent(MetaMetricsEventName.AccountAdded);
+          } catch (e) {
+            // Error occurred while naming the account
+            const error = (e as Error).message;
+
+            await showError(
+              controllerMessenger,
+              snapId,
+              {
+                icon: IconName.UserCircleAdd,
+                title: t('snapAccountNamingFailed'),
+              },
+              {
+                message: t(
+                  'snapAccountNamingFailedDescription',
+                  snapName,
+                ) as string,
+                learnMoreLink,
+                error,
+              },
+            );
+
+            trackSnapAccountEvent(MetaMetricsEventName.AccountNameFailed);
+
+            throw new Error(
+              `Error occurred while naming snap account: ${error}`,
+            );
           }
-          // End the approval flow if it was started
-          if (!skipConfirmation) {
-            controllerMessenger.call('ApprovalController:endFlow', {
-              id: confirmationApprovalId,
-            });
-          }
+        } else {
+          // User has cancelled account creation so remove the account from the keyring
+          await removeAccountHelper(address);
+          await handleUserInput(accountNameConfirmationResult);
+          await persistKeyringHelper();
```

As discussed [here](https://github.com/MetaMask/metamask-extension/pull/25191#issuecomment-2183006307), we should throw an error here too, to keep the same original behavior:

```suggestion
          await persistKeyringHelper();
          
          throw new Error('User denied account creation');
```